types/shapediver/appbuilderinteractiontypes.ts

```diff
@@ -0,0 +1,106 @@
+export type InteractionParameterType = "selection" | "dragging" | "gumball";
+
+/**
+ * PART 1 - Parameter Definitions
+ * 
+ * These are the properties that are defined in GH.
+ * These definitions will in the future live in the Viewer.
+ */
+
+/**
+ * General properties of an interaction parameter.
+ */
+export interface IGeneralInteractionParameterProperties {
+    /** If the objects are hoverable. (default: true) */
+    hover?: boolean,
+    /** The names of the objects that can be interacted with. (see Jira document and discussion result) */
+    nameFilter?: string[]
+}
+
+/**
+ * Properties of a selection parameter.
+ */
+export interface IInteractionParameterPropsSelection extends IGeneralInteractionParameterProperties {
+    /** The minimum number of objects that can be selected. (default: 1) */
+    minimumSelection?: number,
+    /** The maximum number of objects that can be selected. (default: 1) */
+    maximumSelection?: number,
+}
+
+/**
+ * Properties of a dragging parameter.
+ * 
+ * This is a placeholder for now.
+ */
+export interface IInteractionParameterPropsDragging extends IGeneralInteractionParameterProperties {}
+
+/**
+ * Properties of a gumball parameter.
+ */
+export interface IInteractionParameterPropsGumball extends IGeneralInteractionParameterProperties {
+    /** The initial frame of the gumball. */
+    initialFrame?: any, // TODO: How to define this?
```

Comment:
@MajorMeerkatThe3rd it would be sufficient to use an array of two 3-dimensional vectors. These vectors are expected to be orthonormal. Their cross-product defines the third axis.

Reply:
default [[1,0,0],[0,1,0]]

Reply:
@MajorMeerkatThe3rd are you referring to the initial scaling factors of the gumball control? I would always keep this set to `1`. The vectors defining the initial frame can be assumed to be unitized. @mathieu-shapediver do you agree?

Reply:
@snabela @mathieu-shapediver added two new properties instead of the initial frame, `initialVectorU` and `initialVectorV`. We have a vectorU and vectorV also for the drawing tools plane definition, so I like the consistency.